diff --git a/node_modules/@featherbear/presonus-studiolive-api/.DS_Store b/node_modules/@featherbear/presonus-studiolive-api/.DS_Store
new file mode 100644
index 0000000..8b510fc
Binary files /dev/null and b/node_modules/@featherbear/presonus-studiolive-api/.DS_Store differ
diff --git a/node_modules/@featherbear/presonus-studiolive-api/dist/_internal.cjs b/node_modules/@featherbear/presonus-studiolive-api/dist/_internal.cjs
index 600453f..a52cb8a 100644
--- a/node_modules/@featherbear/presonus-studiolive-api/dist/_internal.cjs
+++ b/node_modules/@featherbear/presonus-studiolive-api/dist/_internal.cjs
@@ -1074,6 +1074,9 @@ let counts;
 function setCounts(channelCount) {
 	counts = channelCount;
 }
+
+
+/* 
 function parseChannelString(selector, whitelist) {
 	let { type, channel } = selector;
 	const { mixType, mixNumber } = selector;
@@ -1093,6 +1096,64 @@ function parseChannelString(selector, whitelist) {
 	channel = Math.trunc(channel);
 	if (!(channel > 0) || channel !== oldChannel || counts && !(counts[type] && counts[type] >= channel)) throw new Error("Invalid channel provided");
 	return `${Channel[type]}/ch${channel}`;
+}*/
+
+
+/**
+ * Parse Channel String
+ * @author Max Miller
+ * @param {ChannelSelector} selector 
+ * @param {boolean} whitelist 
+ * @returns {String} channel selection string for SL API
+ */
+function parseChannelString(selector, whitelist) {
+	let { chanType, channelNum } = selector;
+	const { mixType, mixNumber } = selector;
+	
+	// Check if there's a valid mix type flag to select in the specific channel
+	if (counts && mixType) {
+		if (!mixNumber || mixNumber < 1) {
+			throw new Error("Invalid mix channel provided");
+			return "";
+		}
+		
+		if (mixType == "AUX" || mixType == "FX") {
+			if (mixNumber > counts[mixType]) {
+				throw new Error("Invalid mix channel provided");
+				return "";
+			}
+			// Valid AUX or FX mix selection
+		}
+		// Allow Selecting a line channel if the channel type is a DCA
+		else if (chanType == "DCA" && mixType == "LINE") {
+
+		}
+		else {
+			throw new Error("Unexpected mix type");
+		}
+	}
+
+	// Check if chanType is valid
+	if (!Object.keys(Channel).includes(chanType)) {
+		throw new Error("Invalid channel type provided")
+	}
+
+	// If selecting MAIN or TALKBACK, force channelNum to be 1
+	if (["MAIN", "TALKBACK"].includes(chanType)) {
+		channelNum = 1;
+	}
+
+
+	const oldChannel = channelNum;
+	channelNum = Math.trunc(channelNum);
+
+	// Make sure channel actually exsists inside of the chanType
+	if (channelNum <= 0 || channelNum !== oldChannel || counts && !(counts[chanType] && counts[chanType] >= channelNum)) {
+		throw new Error("Invalid channel provided");
+	}
+
+	// return the selector for the channel
+	return `${Channel[chanType]}/ch${channelNum}`;
 }
 //#endregion
 //#region src/lib/util/jsonPacketUtil.ts
@@ -1703,6 +1764,55 @@ var Client = class {
 	getColour(...args) {
 		return this.getColor.apply(this, args);
 	}
+
+	/**
+	 * Set Channel Name
+	 * @author Max Miller
+	 * @param {ChannelSelector} selector 
+	 * @param {String} name 
+	 */
+	setName(selector, name) {
+		this._sendPacket(MessageCode.ParamChars, Buffer.concat([
+			Buffer.from(`${parseChannelString(selector)}/name`),
+			Buffer.from([name])
+		]));
+	}
+
+	/**
+	 * Assign DCA
+	 * @author Max Miller
+	 * @param {import("@featherbear/presonus-studiolive-api").ChannelSelector} selector 
+	 * @param {Int[]} assign_chans 
+	 */
+	assign_dca(selector, assign_chans) {
+		if (selector.type !== "DCA") {
+			throw new Error("Invalid Channel Type!")
+			return;
+		}
+
+		const targetChanString = parseChannelString(selector);
+		
+		// Loop through every line in the DCA
+		for (let n = 1; n <= this.channelCounts["LINE"]; n++) {
+			const path = `${targetChanString}/line${n}`;
+			const cur_state = this.state.get(path);
+
+			// If it's in assigned channels, we should assign it
+			if (n in assign_chans && cur_state !== 1) {
+				this._sendPacket(MessageCode.ParamValue, Buffer.concat([Buffer.from(targetString + "\0\0\0"), 1]));
+			}
+			// Otherwise unassign it
+			else if (!(n in assign_chans) && cur_state !== 0) {
+				this._sendPacket(MessageCode.ParamValue, Buffer.concat([Buffer.from(targetString + "\0\0\0"), 0]));
+			}
+			
+			// Otherwise it must already be correct
+			else {
+				// console.log("[presonus api]: DCA correctly assigned already!");
+			}
+		}
+	}
+
 	/**
 	* For a mono channel, the pan value is the pan value from 0 (hard left) to 100 (hard right)
 	* For a stereo channel, the pan value is the width from 0 to 100 (stereo)
diff --git a/node_modules/@featherbear/presonus-studiolive-api/dist/_internal.mjs b/node_modules/@featherbear/presonus-studiolive-api/dist/_internal.mjs
index eae9828..54d99ea 100644
--- a/node_modules/@featherbear/presonus-studiolive-api/dist/_internal.mjs
+++ b/node_modules/@featherbear/presonus-studiolive-api/dist/_internal.mjs
@@ -1050,6 +1050,7 @@ let counts;
 function setCounts(channelCount) {
 	counts = channelCount;
 }
+/*
 function parseChannelString(selector, whitelist) {
 	let { type, channel } = selector;
 	const { mixType, mixNumber } = selector;
@@ -1069,6 +1070,64 @@ function parseChannelString(selector, whitelist) {
 	channel = Math.trunc(channel);
 	if (!(channel > 0) || channel !== oldChannel || counts && !(counts[type] && counts[type] >= channel)) throw new Error("Invalid channel provided");
 	return `${Channel[type]}/ch${channel}`;
+}*/
+
+
+/**
+ * Parse Channel String
+ * @author Max Miller
+ * @param {import("@featherbear/presonus-studiolive-api").ChannelSelector} selector 
+ * @param {boolean} whitelist 
+ * @returns {String} channel selection string for SL API
+ */
+function parseChannelString(selector, whitelist) {
+	let { chanType, channelNum } = selector;
+	const { mixType, mixNumber } = selector;
+	
+	// Check if there's a valid mix type flag to select in the specific channel
+	if (counts && mixType) {
+		if (!mixNumber || mixNumber < 1) {
+			throw new Error("Invalid mix channel provided");
+			return "";
+		}
+		
+		if (mixType == "AUX" || mixType == "FX") {
+			if (mixNumber > counts[mixType]) {
+				throw new Error("Invalid mix channel provided");
+				return "";
+			}
+			// Valid AUX or FX mix selection
+		}
+		// Allow Selecting a line channel if the channel type is a DCA
+		else if (chanType == "DCA" && mixType == "LINE") {
+			// Allow
+		}
+		else {
+			throw new Error("Unexpected mix type");
+		}
+	}
+
+	// Check if chanType is valid
+	if (!Object.keys(Channel).includes(chanType)) {
+		throw new Error("Invalid channel type provided")
+	}
+
+	// If selecting MAIN or TALKBACK, force channelNum to be 1
+	if (["MAIN", "TALKBACK"].includes(chanType)) {
+		channelNum = 1;
+	}
+
+
+	const oldChannel = channelNum;
+	channelNum = Math.trunc(channelNum);
+
+	// Make sure channel actually exsists inside of the chanType
+	if (channelNum <= 0 || channelNum !== oldChannel || counts && !(counts[chanType] && counts[chanType] >= channelNum)) {
+		throw new Error("Invalid channel provided");
+	}
+
+	// return the selector for the channel
+	return `${Channel[chanType]}/ch${channelNum}`;
 }
 //#endregion
 //#region src/lib/util/jsonPacketUtil.ts
@@ -1626,7 +1685,7 @@ var Client = class {
 	* @private
 	*/
 	_getMuteTargetString(selector) {
-		let targetString = parseChannelString(selector);
+		let targetString = parseChannelString(selector);  // "{Channel[type]}/ch{channel}"
 		if (selector.mixType) targetString += `/assign_${selector.mixType.toLowerCase()}${selector.mixNumber}`;
 		else targetString += "/mute";
 		return targetString;
@@ -1679,6 +1738,60 @@ var Client = class {
 	getColour(...args) {
 		return this.getColor.apply(this, args);
 	}
+
+	/**
+	 * Set Channel Name
+	 * @author Max Miller
+	 * @param {ChannelSelector} selector 
+	 * @param {String} name 
+	 */
+	setName(selector, name) {
+		this._sendPacket(MessageCode.ParamChars, Buffer.concat([
+			Buffer.from(`${parseChannelString(selector)}/name\x00\x00\x00`),
+			Buffer.from(name)
+		]));
+	}
+
+	/**
+	 * Assign DCA
+	 * @author Max Miller
+	 * @param {import("@featherbear/presonus-studiolive-api").ChannelSelector} selector 
+	 * @param {Int[]} assign_chans 
+	 * @param {Int[]}
+	 */
+	assign_dca(selector, assign_chans, controlled_chans = null) {
+		if (selector.type !== "DCA") {
+			throw new Error("Invalid Channel Type!")
+			return;
+		}
+
+		const targetChanString = parseChannelString(selector);
+
+		if (controlled_chans == null) {
+			controlled_chans = Array.from({ length: this.channelCounts["LINE"] }, (_, index) => index + 1);
+		} 
+		
+		// Loop through every line in the DCA
+		for (n in controlled_chans) {
+			const path = `${targetChanString}/line${n}`;
+			const cur_state = this.state.get(path);
+
+			// If it's in assigned channels, we should assign it
+			if (n in assign_chans && cur_state !== 1) {
+				this._sendPacket(MessageCode.ParamValue, Buffer.concat([Buffer.from(targetString + "\0\0\0"), 1]));
+			}
+			// Otherwise unassign it
+			else if (!(n in assign_chans) && cur_state !== 0) {
+				this._sendPacket(MessageCode.ParamValue, Buffer.concat([Buffer.from(targetString + "\0\0\0"), 0]));
+			}
+			
+			// Otherwise it must already be correct
+			else {
+				// console.log("[presonus api]: DCA correctly assigned already!");
+			}
+		}
+	}
+
 	/**
 	* For a mono channel, the pan value is the pan value from 0 (hard left) to 100 (hard right)
 	* For a stereo channel, the pan value is the width from 0 to 100 (stereo)
@@ -1705,27 +1818,52 @@ var Client = class {
 	setLink(selector, link) {
 		this._sendPacket(MessageCode.ParamValue, Buffer.concat([Buffer.from(`${parseChannelString(selector)}/link\x00\x00\x00`), toBoolean(link)]));
 	}
+
+	/**
+	 * Get Level Specfic Selector String
+	 * @author Max Miller
+	 * @param {import("@featherbear/presonus-studiolive-api").ChannelSelector} selector 
+	 * @returns 
+	 */
 	_getLevelString(selector) {
 		let targetString = parseChannelString(selector);
-		if (selector.mixType) switch (selector.mixType) {
-			case "AUX":
+		
+		if (selector.mixType) {
+			if (selector.mixType == "AUX") {
 				targetString += `/${Channel[selector.mixType]}${selector.mixNumber}`;
-				break;
-			case "FX":
+			}
+
+			else if (selector.mixType == "FX") {
 				targetString += `/FX${String.fromCharCode(64 + selector.mixNumber)}`;
-				break;
-			default: throw new Error("Unexpected mix type");
+			}
+
+			// No volume adjustment needed, so mixtype LINE is not allowed here
+
+			else {
+				throw new Error("Unexpected mix type");
+			}
 		}
-		else targetString += "/volume";
+		else {
+			targetString += "/volume";
+		}
+
 		return targetString;
 	}
+
 	getLevel(selector) {
 		return this.state.get(this._getLevelString(selector));
 	}
+	
 	/**
-	* @internal Send a level command to the target
-	* targetLevel - [0, 100]
-	*/
+	 * Set Level
+	 * targetLevel - [0, 100]
+	 * @author Max Miller
+	 * @internal Send a level command to the target
+	 * @param {import("@featherbear/presonus-studiolive-api").ChannelSelector} selector 
+	 * @param {Float} targetLevel 
+	 * @param {number} duration 
+	 * @returns {Promise}
+	 */
 	_setLevel(selector, targetLevel, duration = 0) {
 		const targetString = this._getLevelString(selector);
 		const assertReturn = () => {
@@ -1751,6 +1889,7 @@ var Client = class {
 			});
 		});
 	}
+
 	/**
 	* Set volume (decibels)
 	*
@@ -1760,6 +1899,7 @@ var Client = class {
 	async setChannelVolumeLogarithmic(selector, decibel, duration) {
 		return this._setLevel(selector, logVolumeToLinear(decibel), duration);
 	}
+
 	/**
 	* Set volume (pseudo intensity)
 	*
@@ -1772,6 +1912,7 @@ var Client = class {
 	async setChannelVolumeLinear(selector, linearLevel, duration) {
 		return this._setLevel(selector, linearLevel, duration);
 	}
+
 	/**
 	* Look at metering data and adjust channel fader so that the level is of a certain loudness
 	* NOTE: This is not perceived loudness. Not very practical, but useful in a pinch?
diff --git a/node_modules/@featherbear/presonus-studiolive-api/dist/types/lib/Client.d.ts b/node_modules/@featherbear/presonus-studiolive-api/dist/types/lib/Client.d.ts
index 042b10e..0c1868c 100644
--- a/node_modules/@featherbear/presonus-studiolive-api/dist/types/lib/Client.d.ts
+++ b/node_modules/@featherbear/presonus-studiolive-api/dist/types/lib/Client.d.ts
@@ -153,6 +153,23 @@ export declare class Client {
     setColour(...args: Parameters<this["setColor"]>): any;
     getColor(selector: ChannelSelector): any;
     getColour(...args: Parameters<this["getColor"]>): any;
+
+    /**
+	 * Set Channel Name
+	 * @author Max Miller
+	 * @param {ChannelSelector} selector 
+	 * @param {String} name 
+	 */
+	setName(selector: ChannelSelector, name: string): void;
+
+    /**
+	 * Assign DCA
+	 * @author Max Miller
+	 * @param {import("@featherbear/presonus-studiolive-api").ChannelSelector} selector 
+	 * @param {Int[]} assign_chans 
+	 */
+	assign_dca(selector: ChannelSelector, assign_chans: Int[]): void;
+
     /**
      * For a mono channel, the pan value is the pan value from 0 (hard left) to 100 (hard right)
      * For a stereo channel, the pan value is the width from 0 to 100 (stereo)
diff --git a/node_modules/@featherbear/presonus-studiolive-api/dist/types/lib/types/ChannelSelector.d.ts b/node_modules/@featherbear/presonus-studiolive-api/dist/types/lib/types/ChannelSelector.d.ts
index 3f38500..570dc18 100644
--- a/node_modules/@featherbear/presonus-studiolive-api/dist/types/lib/types/ChannelSelector.d.ts
+++ b/node_modules/@featherbear/presonus-studiolive-api/dist/types/lib/types/ChannelSelector.d.ts
@@ -1,8 +1,11 @@
 import type { ChannelTypes } from "../constants";
 type MixSelector = {
-    mixType: ChannelTypes & ("AUX" | "FX");
+    // NOTE: ONLY ALLOW LINE WHEN SELECTING A DCA
+    mixType: ChannelTypes & ("AUX" | "FX" | "LINE");
     mixNumber: number;
 };
+
+
 type ChannelSelector = ({
     type: ChannelTypes;
     channel: number;
